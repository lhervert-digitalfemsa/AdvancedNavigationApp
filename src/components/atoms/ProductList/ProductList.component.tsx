import React, { useContext } from "react";
import { View, Text, FlatList, TextInput } from 'react-native';
import Product from "../Product/Product.component";
import { AppContextType } from "../../../hooks/useContext";
import { styles } from "./ProductList.styles";

const ProductList = ({ products }: { products: AppContextType["products"] }) => {

  return (
    <View>
      <FlatList
        data={products}
        contentContainerStyle={styles.container}
        renderItem={({ item }) => <Product {...item} />}
        keyExtractor={item => item.id + ''}
      />
    </View>
  );
}

export default ProductList;