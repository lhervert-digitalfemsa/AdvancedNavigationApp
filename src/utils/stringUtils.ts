export function capitalizeFirstLetter(string: string | undefined): string {
  if (!string) {
    return '';
  } else {
    return string.charAt(0).toUpperCase() + string.slice(1);
  }
}
