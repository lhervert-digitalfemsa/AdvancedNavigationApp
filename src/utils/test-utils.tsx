import {
  act,
  cleanup,
  fireEvent,
  render,
  RenderOptions,
  waitFor,
} from '@testing-library/react-native';
import { ApplicationProvider } from '@ui-kitten/components';
import React, { ReactElement, ReactNode } from 'react';
import * as eva from '@eva-design/eva';
import { NavigationContainer } from '@react-navigation/native';
import AppContext, { AppContextType } from '../hooks/useContext';
import { ProductT } from '../types/Product.type';

export const mockContextValues = {
  products: [
    { id: 1, title: 'Apple', description: 'A red apple', price: 1.0 },
    { id: 2, title: 'Banana', description: 'A yellow banana', price: 0.5 },
    { id: 3, title: 'Cherry', description: 'A sweet cherry', price: 2.0 },
  ],
  cart: [],
  loading: false,
  isUserLoggedIn: true,
  user: {
    name: 'John Doe',
    token: '1234',
  },
  error: undefined,
  settings: {
    theme: 'light' as 'light' | 'dark',
  },
  saveSettings: (settings: AppContextType['settings']) => {

  },
  login: (user: AppContextType['user']) => {

  },
  logout: () => {

  },
  addToCart: (product: ProductT) => { },
  removeFromCart: (productId: number) => { },
  updateCartItem: (productId: number, quantity: number) => { },
  clearCart: () => { },
}


const AllProviders = ({ children }: { children: ReactNode }) => {
  return (
    <ApplicationProvider {...eva} theme={eva.dark}>
      <NavigationContainer>
        <AppContext.Provider value={mockContextValues}>
          {children}
        </AppContext.Provider>
      </NavigationContainer>
    </ApplicationProvider>
  )
};

const customRender = (ui: ReactElement, options?: RenderOptions) =>
  render(ui, { wrapper: AllProviders, ...options });

export { customRender as render, fireEvent, act, waitFor, cleanup };
