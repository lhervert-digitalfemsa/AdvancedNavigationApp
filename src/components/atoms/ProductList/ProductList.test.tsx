import React from 'react';
import { render } from '../../../utils/test-utils';
import ProductList from './ProductList.component';
import { ApplicationProvider } from '@ui-kitten/components';
import * as eva from '@eva-design/eva';
import AppContext from "../../../hooks/useContext";

jest.mock('@react-native-async-storage/async-storage', () =>
  require('@react-native-async-storage/async-storage/jest/async-storage-mock')
);

describe('ProductList', () => {
  const mockProducts = [
    { id: 1, title: 'Apple', description: 'A red apple', price: 1.0 },
    { id: 2, title: 'Banana', description: 'A yellow banana', price: 0.5 },
    { id: 3, title: 'Cherry', description: 'A sweet cherry', price: 2.0 }
  ];

  it('renders a list of products', () => {
    const { getByText } = render(
      <ProductList products={mockProducts} />
    );

    expect(getByText('Apple')).toBeTruthy();
    expect(getByText('Banana')).toBeTruthy();
    expect(getByText('Cherry')).toBeTruthy();
  });

  it('renders no products when given an empty array', () => {
    const { queryByText } = render(<ProductList products={[]} />);

    expect(queryByText('Apple')).toBeNull();
    expect(queryByText('Banana')).toBeNull();
    expect(queryByText('Cherry')).toBeNull();
  });

  it('matches snapshot', () => {
    const { toJSON } = render(<ProductList products={mockProducts} />);
    expect(toJSON()).toMatchSnapshot();
  });
});
