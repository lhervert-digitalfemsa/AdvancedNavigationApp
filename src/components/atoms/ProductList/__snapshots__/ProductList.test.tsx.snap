// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList matches snapshot 1`] = `
<View>
  <RCTScrollView
    contentContainerStyle={
      {
        "width": "100%",
      }
    }
    data={
      [
        {
          "description": "A red apple",
          "id": 1,
          "price": 1,
          "title": "Apple",
        },
        {
          "description": "A yellow banana",
          "id": 2,
          "price": 0.5,
          "title": "Banana",
        },
        {
          "description": "A sweet cherry",
          "id": 3,
          "price": 2,
          "title": "Cherry",
        },
      ]
    }
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={0.0001}
    stickyHeaderIndices={[]}
    viewabilityConfigCallbackPairs={[]}
  >
    <View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#222B45",
              "borderColor": "#101426",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": undefined,
                "height": 0,
              }
            }
          />
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                {
                  "paddingHorizontal": 24,
                  "paddingVertical": 16,
                },
              ]
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": undefined,
                  }
                }
                style={
                  {
                    "borderRadius": 5,
                    "height": 150,
                    "resizeMode": "contain",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  {
                    "flexShrink": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  appearance="default"
                  category="s1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "600",
                      },
                      undefined,
                    ]
                  }
                >
                  Apple
                </Text>
                <Text
                  appearance="default"
                  category="p1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "400",
                      },
                      {
                        "paddingTop": 20,
                      },
                    ]
                  }
                >
                  $1
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 10,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#3366FF",
                        "borderColor": "#3366FF",
                        "borderRadius": 4,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "minHeight": 40,
                        "minWidth": 40,
                        "opacity": 1,
                        "paddingHorizontal": 10,
                        "paddingVertical": 12,
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="p1"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 15,
                            "fontWeight": "400",
                          },
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "marginHorizontal": 10,
                          },
                        ]
                      }
                    >
                      Add to cart
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#222B45",
              "borderColor": "#101426",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": undefined,
                "height": 0,
              }
            }
          />
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                {
                  "paddingHorizontal": 24,
                  "paddingVertical": 16,
                },
              ]
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": undefined,
                  }
                }
                style={
                  {
                    "borderRadius": 5,
                    "height": 150,
                    "resizeMode": "contain",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  {
                    "flexShrink": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  appearance="default"
                  category="s1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "600",
                      },
                      undefined,
                    ]
                  }
                >
                  Banana
                </Text>
                <Text
                  appearance="default"
                  category="p1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "400",
                      },
                      {
                        "paddingTop": 20,
                      },
                    ]
                  }
                >
                  $0.5
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 10,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#3366FF",
                        "borderColor": "#3366FF",
                        "borderRadius": 4,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "minHeight": 40,
                        "minWidth": 40,
                        "opacity": 1,
                        "paddingHorizontal": 10,
                        "paddingVertical": 12,
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="p1"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 15,
                            "fontWeight": "400",
                          },
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "marginHorizontal": 10,
                          },
                        ]
                      }
                    >
                      Add to cart
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        onFocusCapture={[Function]}
        onLayout={[Function]}
        style={null}
      >
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "backgroundColor": "#222B45",
              "borderColor": "#101426",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "opacity": 1,
              "overflow": "hidden",
            }
          }
        >
          <View
            style={
              {
                "backgroundColor": undefined,
                "height": 0,
              }
            }
          />
          <View
            style={
              [
                {
                  "flexShrink": 1,
                },
                {
                  "paddingHorizontal": 24,
                  "paddingVertical": 16,
                },
              ]
            }
          >
            <View
              style={
                {
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  {
                    "uri": undefined,
                  }
                }
                style={
                  {
                    "borderRadius": 5,
                    "height": 150,
                    "resizeMode": "contain",
                    "width": 100,
                  }
                }
              />
              <View
                style={
                  {
                    "flexShrink": 1,
                    "paddingHorizontal": 10,
                  }
                }
              >
                <Text
                  appearance="default"
                  category="s1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "600",
                      },
                      undefined,
                    ]
                  }
                >
                  Cherry
                </Text>
                <Text
                  appearance="default"
                  category="p1"
                  status="basic"
                  style={
                    [
                      {
                        "color": "#FFFFFF",
                        "fontFamily": "System",
                        "fontSize": 15,
                        "fontWeight": "400",
                      },
                      {
                        "paddingTop": 20,
                      },
                    ]
                  }
                >
                  $2
                </Text>
                <View
                  style={
                    {
                      "alignItems": "center",
                      "display": "flex",
                      "flexDirection": "row",
                      "justifyContent": "center",
                      "marginTop": 10,
                    }
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#3366FF",
                        "borderColor": "#3366FF",
                        "borderRadius": 4,
                        "borderWidth": 1,
                        "flexDirection": "row",
                        "justifyContent": "center",
                        "minHeight": 40,
                        "minWidth": 40,
                        "opacity": 1,
                        "paddingHorizontal": 10,
                        "paddingVertical": 12,
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="p1"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 15,
                            "fontWeight": "400",
                          },
                          {
                            "color": "#FFFFFF",
                            "fontFamily": "System",
                            "fontSize": 14,
                            "fontWeight": "bold",
                            "marginHorizontal": 10,
                          },
                        ]
                      }
                    >
                      Add to cart
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
